sdk/clientcore/core/src/main/java/io/clientcore/core/utils/ServerSentEventUtils.java

```diff
@@ -112,18 +110,106 @@ private static boolean isEndOfBlock(StringBuilder sb) {
     /**
      * Processes the SSE buffer and dispatches the event.
      *
-     * @param reader The {@link BufferedReader} to read data from.
+     * @param inputStream The {@link InputStream} to read data from.
      * @param listener The {@link ServerSentEventListener} object attached to the {@link HttpRequest}.
      */
-    private static ServerSentResult processBuffer(BufferedReader reader, ServerSentEventListener listener) {
+    private static ServerSentResult processBuffer(InputStream inputStream, ServerSentEventListener listener) {
         ServerSentEvent event = null;
         List<String> allCollectedData = new ArrayList<>(); // List to store all collected data blocks
 
         try {
             StringBuilder collectedData = new StringBuilder();
-            String line;
 
-            while ((line = reader.readLine()) != null) {
+            // Use a byte array to accumulate UTF-8 bytes and convert to string line by line
+            List<Byte> lineBytes = new ArrayList<>();
+
+            int b;
+            while ((b = inputStream.read()) != -1) {
```

Comment:
This design is most likely way less efficient as this is reading byte-by-byte and adding it into a list byte-by-byte.